naisplater

```diff
@@ -48,24 +49,43 @@ get_template_files() {
   find "${TEMPLATES_DIR}" "${TEMPLATES_DIR}/${ENV}" -type f -maxdepth 1 2> /dev/null
 }
 
+decrypt() {
```

Comment:
It doesn't seem to work with maps, or i might be using it the wrong way.

Template:
```yaml
{{ .mysecret }} is the secret
{{ .map.key }} is the secret in the map
```
Vars:
```yaml
mysecret.enc: U2FsdGVkX1/9jAP74brTRyrhzac4LvAFaDdTRef1TGs=
map:
  key.enc: U2FsdGVkX1/9jAP74brTRyrhzac4LvAFaDdTRef1TGs=
```
Output:
```bash
/bin/bash naisplater dev ./test/templates ./test/vars ./test/out supersecret69 > /dev/null
error reading input file
unable to decrypt:   key.enc: U2FsdGVkX1/9jAP74brTRyrhzac4LvAFaDdTRef1TGs=
2018/07/18 12:20:02 Failed to parse standard input: yaml: line 6: mapping values are not allowed in this context
make: *** [test] Error 1
```

Reply:
No, you're not doing anything wrong - it's just not supported (should be clarified in docs). When doing decryption it's not yaml-aware, and just reads the vars-file line by line. So maps, or even nesting will not work.
I think this might be an acceptable limitation until we rewrite.

Reply:
Planlegger vi en omskrivning? Til hva?